sqlglot/dialects/tsql.py

```diff
@@ -365,6 +368,57 @@ class Parser(parser.Parser):
 
         CONCAT_NULL_OUTPUTS_STRING = True
 
+        def _parse_commit_or_rollback(self) -> exp.Commit | exp.Rollback:
+            """Applies to SQL Server and Azure SQL Database
+            COMMIT [ { TRAN | TRANSACTION }
+                [ transaction_name | @tran_name_variable ] ]
+                [ WITH ( DELAYED_DURABILITY = { OFF | ON } ) ]
+                [ ; ]
+
+            ROLLBACK { TRAN | TRANSACTION }
+                [ transaction_name | @tran_name_variable
+                | savepoint_name | @savepoint_variable ]
+                [ ; ]
+            """
+            rollback = self._prev.token_type == TokenType.ROLLBACK
+
+            self._match_texts({"TRAN", "TRANSACTION"})
+            txn_name = self._parse_id_var()
+
+            durability = None
+            if self._match_pair(TokenType.WITH, TokenType.L_PAREN):
+                self._match_text_seq("DELAYED_DURABILITY")
+                self._match(TokenType.EQ)
+                if self._match_text_seq("OFF"):
+                    durability = False
+                else:
+                    self._match(TokenType.ON)
+                    durability = True
+
+                self._match_r_paren()
+
+            if rollback:
+                return self.expression(exp.Rollback, this=txn_name)
+
+            return self.expression(exp.Commit, this=txn_name, durability=durability)
+
+        def _parse_transaction(self) -> exp.Transaction | exp.Command:
+            """Applies to SQL Server and Azure SQL Database
+            BEGIN { TRAN | TRANSACTION }
+            [ { transaction_name | @tran_name_variable }
+            [ WITH MARK [ 'description' ] ]
+            ]
+            [ ; ]
+            """
+            if self._match_texts(("TRAN", "TRANSACTION")):
+                # we have a transaction and not a BEGIN
+                transaction = self.expression(exp.Transaction, this=self._parse_id_var())
+                if self._match_text_seq("WITH", "MARK"):
+                    transaction.set("mark", self._parse_string())
+                return transaction
+
+            return self.expression(exp.Command, this="BEGIN")
```

Comment:
This should instead be:

```suggestion
            return self._parse_as_command(self._prev)
```

The `exp.Command` class is generally used as a fallback and just stores a raw string from the "command" token to the end of the chunk being parsed. This allows the parser to be lenient instead of failing, even in cases where we don't actually parse the syntax of a dialect-specific command.

Implementing this suggestion covers what you've already tested + more, so it's more robust.

Reply:
Done.